DESIGN.md

```diff
@@ -0,0 +1,63 @@
+
+### Depot
+
+The 'WNFS' file system that holds our user's data consists out of [IPLD](https://ipld.io/) blocks. These blocks make up the content-adressable web, another core idea to the ODD stack, making our data tamper-proof.
+
+This component is reponsible for retrieving and storing these blocks. On the lowest level this means for any given bytes (data) give me back a CID (content address). And vice versa, when I present a CID, give me back the bytes for it.
+
+Looking at this from a higher viewpoint, this unlocks the larger part of data syncing, we can transfer our blocks to a remote source.
+
+### Account
+
+Most of the time to push the data to a remote endpoint you will need some sort of account system, otherwise anyone could fill up the external storage space which costs money. The user accounts will be represented through authorisation, UCANs, our authorisation tickets.
+
+In order to use ts-odd you'll be creating a `Program`, this is a specific setup of components, an artificial environment if you will. This program has a UCAN store, a repository of "authorisation tickets" collected from various sources. The account system is one of those sources. It can also access that UCAN store and then use it to authorise various actions.
+
+This may also tie into the other part of data syncing, taking the file system's root CID (aka. data root) and storing it somewhere the other devices can reach it.
+
+### Identifier & Agent
+
+The account system also involves the identifier and agent components. The identifier signifies one of the user's identifiers and the agent delegates to external services. Both of these use decentralised identifiers ([DID](https://www.w3.org/TR/did-core/)s).
+
+Typically the agent will use a temporary session key pair and the identifier is the more permanent identifier. Example flow: Identifier delegates to agent, agent contacts remote account service, account service issues UCANs addressed to identifier. Those UCANs are then used throughout the SDK to check for capabilities, etc.
+
+The reason we have two components here is so that the identifier doesn't need to sign every time. This is useful for example with passkeys or blockchain wallets where you would get a popup each time you'd need to sign something.
+
+### Authority
+
+This component is reponsible for providing and requesting authority. Which technically means providing and requesting UCANs and file system secrets (access keys).
+
+### Channel
+
+The channel component is used to establish a channel. It serves as the public channel for the [AWAKE protocol](https://github.com/ucan-wg/awake). You can use whatever channel here as the AWAKE protocol used on top is responsible for establish a secure session. Typically this will be websockets or a P2P channel.
```

Comment:
Explain AWAKE briefly; below there's a very brief explanation, but expand and put the use-case before the description of the mechanism